e2e-tests/src/main.rs

```diff
@@ -0,0 +1,235 @@
+    // session period is set to 5 blocks (see `run_consensus.sh`), plus one to be on the safe site
+    wait_for_finalized_block(connection, 6)
+}
+
+fn test_token_transfer(config: Config) -> anyhow::Result<()> {
+    let Config { node, seeds, .. } = config;
+
+    let accounts: Vec<sr25519::Pair> = match seeds {
+        Some(seeds) => seeds
+            .into_iter()
+            .map(|seed| {
+                sr25519::Pair::from_string(&seed, None).expect("Can't create pair from seed value")
+            })
+            .collect(),
+        None => vec!["//Damian", "//Tomasz", "//Zbyszko", "//Hansu"]
+            .iter()
+            .map(|seed| {
+                sr25519::Pair::from_string(seed, None).expect("Can't create pair from seed value")
+            })
+            .collect(),
+    };
+
+    let from: sr25519::Pair = accounts.get(0).expect("No accounts passed").to_owned();
+
+    let to = AccountId::from(
+        accounts
+            .get(1)
+            .expect("Pass at least two accounts")
+            .public(),
+    );
+
+    let connection = create_connection(format!("ws://{}", node)).set_signer(from);
+
+    let balance_before = connection
+        .get_account_data(&to)?
+        .expect("Could not get account data")
+        .free;
+
+    info!("[+] Account {} balance before tx: {}", to, balance_before);
+
+    let transfer_value = 1000u128;
+
+    let tx: UncheckedExtrinsicV4<_> = compose_extrinsic!(
+        connection,
+        "Balances",
+        "transfer",
+        GenericAddress::Id(to.clone()),
+        Compact(transfer_value)
+    );
+
+    // send and watch extrinsic until InBlock
+    let tx_hash = connection
+        .send_extrinsic(tx.hex_encode(), XtStatus::InBlock)?
+        .expect("Could not get tx hash");
+
+    info!("[+] Transaction hash: {}", tx_hash);
+
+    let balance_after = connection
+        .get_account_data(&to)?
+        .expect("Could not get account data")
+        .free;
+
+    info!("[+] Account {} balance after tx: {}", to, balance_after);
+
+    assert!(
+        balance_before + transfer_value == balance_after,
+        "before = {}, after = {}, tx = {}",
+        balance_before,
+        balance_after,
+        transfer_value
+    );
+
+    Ok(())
+}
+
+fn test_change_validators(config: Config) -> anyhow::Result<()> {
+    let Config { node, seeds, sudo } = config;
+
+    let accounts = accounts(seeds);
+
+    let sudo = match sudo {
+        Some(seed) => {
+            sr25519::Pair::from_string(&seed, None).expect("Cant create Pair from seed value")
+        }
+        None => accounts.get(0).expect("whoops").to_owned(),
+    };
+
+    let connection = create_connection(format!("ws://{}", node)).set_signer(sudo);
+
+    let validators_before: Vec<AccountId> = connection
+        .get_storage_value("Session", "Validators", None)?
+        .unwrap();
+
+    info!("[+] Validators before tx: {:#?}", validators_before);
+
+    let mut new_validators: Vec<AccountId> = accounts
```

Comment:
We can easily get rid of this nasty `mut` by `.chain`ing the iterator with `iter::once(AccountId::from_ss58check(...).unwrap())` before collecting

Reply:
brilliant, thanks!